### ISSUE TITLE
Issue compiling with {{else if}} in inline partials

### ISSUE THREAD

**Issue by cjbarth** [zordius/lightncandy#303]
The PHP Code:
-------------

```php
require('./vendor/autoload.php');
use LightnCandy\LightnCandy;

// The Template:
$template = <<<VAREND
{{#*inline "t1"}}
    {{#if url}}
        <a />
    {{else if imageUrl}}
        <img />
    {{else}}
        <span />
    {{/if}}
{{/inline}}
VAREND;

// Helpers:
$helpers = array(
  'foo' => function () {
    return 'Hello World';
  },
);

// Partials:
$partials = array(
    'bar' => 'Yes, partial',
);

$phpStr = LightnCandy::compile($template, array(
  // Used compile flags
  'flags' => LightnCandy::FLAG_ERROR_EXCEPTION | LightnCandy::FLAG_HANDLEBARS | LightnCandy::FLAG_RUNTIMEPARTIAL,
  'helpers' => $helpers,
  'partials' => $partials,
));

echo "Generated PHP Code:\n$phpStr\n";

// Input Data:
$data = array(
  'qoo' => 'moo'
);

// Save the compiled PHP code into a php file
file_put_contents('render.php', '<?php ' . $phpStr . '?>');

// Get the render function from the php file
$renderer = include('render.php');

echo "Result:\n" . $renderer($data);
```

The Issue:
----------

I get the following compilation error:

```PHP Fatal error:  Uncaught Exception: Unexpect token {{/if}} ! Previous token {{#*[inline]}} is not closed in /Users/.../LightnCandy/src/LightnCandy.php:110
```

**Comment by zordius** [comment]
Thank you for your report! Will also check on https://github.com/zordius/lightncandy/pull/302 .

**Comment by zordius** [comment]
Confirmed as a bug, the Error was:
```
Exception: Unexpect token {{/if}} ! Previous token {{#*[inline]}} is not closed
```

**Comment by zordius** [comment]
This should be fixed in 1.2.4 release, please try again, thanks.
